mesheryctl/internal/cli/root/system/channel.go

```diff
@@ -81,11 +82,62 @@ var viewCmd = &cobra.Command{
 	},
 }
 
+//check release channel version string supplied by user for validity. This func does not set any thing.(Just for checks)
+func validVersionCheck(c string) (bool, error) {
+	stableRegx := regexp.MustCompile(`^stable-v[0-9]+\.[0-9]+\.[0-9]+$`)
+	edgeRegx := regexp.MustCompile(`^edge-v[0-9]+\.[0-9]+\.[0-9]+$`)
+
+	if !(stableRegx.MatchString(c) || edgeRegx.MatchString(c)) {
+		return false, fmt.Errorf("misformatted release channel or version")
+	}
+
+	// get mesheryctl config
+	mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+	if err != nil {
+		return false, fmt.Errorf("error processing config %+v", err)
+	}
+	// get current context from mesheryctl config
+	ctxCurrent, err := mctlCfg.GetCurrentContext()
+	if err != nil {
+		return false, fmt.Errorf("error fetching current context %+v", err)
+	}
+
+	verCurrent := ctxCurrent.GetVersion() //store current version
```

Comment:
also would be best if the function would be used as `ctxCurrent.ValidateChannel()`

Reply:
Would that involve writing the invalid version to the central context struct? Let's try to avoid that. I think that would be best.

Reply:
`ValidateVersion` doens't write it's just validate similarly `ValidateChannel` would work.

Reply:
@piyushsingariya  you mean just changing the name of the function?

Reply:
I do not see a `ValidateChannel` method on the context object.

Reply:
I meant changing the name and adding it as method to the context object.